sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/IdentityClient.java

```diff
@@ -640,38 +639,20 @@ private HttpPipeline setupPipeline(HttpClient httpClient) {
      * @return a Publisher that emits an AccessToken
      */
     public Mono<MsalToken> authenticateWithBrowserInteraction(TokenRequestContext request, int port) {
-        String authorityUrl = options.getAuthorityHost().replaceAll("/+$", "") + "/" + tenantId;
-        return AuthorizationCodeListener.create(port)
-            .flatMap(server -> {
-                URI redirectUri;
-                String browserUri;
-                try {
-                    redirectUri = new URI(String.format("http://localhost:%s", port));
-                    browserUri =
-                        String.format("%s/oauth2/v2.0/authorize?response_type=code&response_mode=query&prompt"
-                                + "=select_account&client_id=%s&redirect_uri=%s&state=%s&scope=%s",
-                            authorityUrl,
-                            clientId,
-                            redirectUri.toString(),
-                            UUID.randomUUID(),
-                            String.join(" ", request.getScopes()));
-                } catch (URISyntaxException e) {
-                    return server.dispose().then(Mono.error(e));
-                }
+        URI redirectUri;
+        try {
+            redirectUri = new URI(String.format("http://localhost:%s", port));
+        } catch (URISyntaxException e) {
+            return Mono.error(logger.logExceptionAsError(new RuntimeException(e)));
+        }
+        InteractiveRequestParameters parameters = InteractiveRequestParameters.builder(redirectUri)
+                                                     .scopes(new HashSet<>(request.getScopes()))
+                                                     .build();
+        Mono<IAuthenticationResult> acquireToken = publicClientApplicationAccessor.getValue()
+                               .flatMap(pc -> Mono.fromFuture(() -> pc.acquireToken(parameters)));
```

Comment:
Do you need to use `fromFuture`? Can it just be:
```java
.flatMap(pc -> pc.acquireToken(parameters));
```

Reply:
its needed, because we get back a CompletableFuture.